### ISSUE TITLE
Functional tests flakeyness

### ISSUE THREAD

**Issue by rogerfranklin2** [sky-uk/nowtv-react-toolkit#88]
It's getting pretty annoying when you change the README and the functional tests fail

We have changed to use the fork on nightwatch and it doesn't to seem to have fixed anything
 
What are we going to do about this? Anyone got any solutions that they think might work?

**Comment by tomatau** [comment]
Having a look at the recordings for https://circleci.com/gh/sky-uk/nowtv-react-toolkit/399

The error displayed in browsers is `Unable to connect to page`.

This down to the connection between Cirlce and Browserstack over the tunnel. We faced this before in assurance and our solution was to run the feature tests against a heroku app... avoiding the need to run a localhost server to test against.

**Comment by NowTV-Assurance** [comment]
Started [this](https://github.com/sky-uk/nowtv-react-toolkit/tree/fix-flakey-tests) branch for attempting to debug and hopefully resolve the issue. First idea is the `-forcelocal` flag.

**Comment by tomatau** [comment]
Started [this](https://github.com/sky-uk/nowtv-react-toolkit/tree/fix-flakey-tests) branch for attempting to debug and hopefully resolve the issue. First idea is the `-forcelocal` flag.

**Comment by tomatau** [comment]
Didn't work :( As soon as you uncheck "Resolve all URLs through my network" in the browser stack live GUI, the tests start to fail. This is a really frustrating coupling between the website interface and our tests... and it's not just limited to the toolkit project.

@johnsibly is there any chance of multiple BrowserStack accounts where one can be dedicated to running tests and one for developer debugging through their "live" view?

**Comment by bentidy** [comment]
@tomatau: yes it may well be due to multiple test running for the *same* BrowserStack account. We should use the -localIdentifier command line option of browserstacklocal with a unique identifier to make sure we're not affected by that.

**Comment by tomatau** [comment]
We've been attempting to use that locally with no effect overall. I can literally have the tests pass, then go into the gui, flip the checkbox and rerun the tests to see them fail.

The state of the Browserstack GUI is directly effecting the tests and it shouldn't.

**Comment by alexjfno1** [comment]
@tomatau We have seen a similar thing in our team where we can't connect to the app sometimes.

I can forward you the email address of the support guy at browserstack tomorrow and see if he has any answers.
